src/odb/src/defin/defin.cpp

```diff
@@ -37,70 +37,85 @@
 
 namespace odb {
 
+std::mutex defin::_def_mutex;
+
 defin::defin(dbDatabase* db, utl::Logger* logger, MODE mode)
 {
+  std::lock_guard<std::mutex> lock(_def_mutex);
   _reader = new definReader(db, logger, mode);
 }
 
 defin::~defin()
 {
+  std::lock_guard<std::mutex> lock(_def_mutex);
```

Comment:
I'm not clear the locking is doing anything here. It kind of looks like this is trying to protect two defin's from running at the same time on the same database, but what we're really concerned about is two different defin's running on two different databases.

What are these locks here to do?

Reply:
I think he has just locked every method.  The key ones are ::create*/replace* methods which will block simultaneous parsing.

@LucasYuki I think some of these do nothing more than set a manipulate local variables and not the db and therefore do not need locks (eg ctor, dtor, skip).  Please analyze it a bit more.